### PR TITLE
Text field

### LINKED ISSUE #101: Component > Text Field
Add light/dark theme toggle to text field component with ability to toggle off input and pre-fill with values
Add ability to specify it is a link and open in new tab icon

Screen Shot 2021-10-28 at 8 37 59 PM
Screen Shot 2021-10-28 at 8 57 42 PM

Figma reference: https://www.figma.com/file/vmZ1jRhhLHZDcY6rcrWYT2/Aurora-Design-System?node-id=722%3A87

### PR DESCRIPTION
**Pull request by andykwan1124**
## Description

image

Closes #101 

- The links being purple is intentional, see #101 for screenshots of reqs

### CODE REVIEW

**Comment by andykwan1124** [comment]
Not sure who to request for review here...

File: components/TextField.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+const TextField = ({ darkModeEnabled, isLink, prefillValue, customValue, onChangeCustomValue }) => {
+    return (
+        <StyledTextField darkMode={darkModeEnabled}>
+            {
+                isLink ?
+                <StyledLinkDiv>
+                    <StyledLink href={prefillValue} target='_blank'>
+                        <StyledLinkText>{prefillValue}</StyledLinkText>
+                    </StyledLink>
+                    <NewTabIcon href={prefillValue}/>
```
Comment (kevin-zou) [inline components/TextField.js:76]: Is it possible to use the new `Icon` component for this? It should support FontAwesome icons as well

**Review by kevin-zou — APPROVED** [review]
fire pr, one suggestion to avoid explicitly defining icons in our code

**Review by vincentchiang1 — APPROVED** [review]
good job :))

File: components/TextField.js (hunk 1 above)
Comment (andykwan1124) [reply components/TextField.js]: Yeah that works! I used the `Icon` component instead and the display/behavior is still the same

**Review by kevin-zou — APPROVED** [review]
thanks for making the change, lgtm! ❤️

**Pull request merged**